doc/source/ray-core/patterns/limit-running-tasks.rst

```diff
@@ -0,0 +1,39 @@
+.. _core-patterns-limit-running-tasks:
+
+Pattern: Using resources to limit the number of concurrently running tasks
+==========================================================================
+
+In this pattern, we use :ref:`resources <resource-requirements>` to limit the number of concurrently running tasks.
+
+Running too many tasks at the same time might overload the cluster and cause issues like OOM.
+If that is the case, we can reduce the number of concurrently running tasks by increasing the amount of resources requested by those tasks.
+This works because Ray makes sure that the sum of the resource requirements of all of the concurrently running tasks on a given node cannot exceed the node's total resources.
+
+.. note::
+
+   For actor tasks, the number of actors we create controls the number of concurrently running actor tasks we can have.
+
+Example use case
+----------------
+
+You have a data processing workload that processes each input file independently using Ray :ref:`remote functions <ray-remote-functions>`.
+Since each task needs to load the input data into heap memory and do the processing, running too many of them can cause OOM.
+In this case, you can use the ``memory`` resource to limit the number of concurrently running tasks (usage of other resources like ``num_cpus`` can achieve the same goal as well).
```

Comment:
Should probably add a note to clarify that memory is a logical resource.

Also, it would be nice to link or mention @clarng's ongoing memory monitor work.